index.js

```diff
@@ -0,0 +1,238 @@
+  } else {
+    const matr = matrices[chatId][name];
+    const invertedMatr = matr.getInvert();
+    const text = typeof invertedMatr === 'string' ?
+      invertedMatr :
+      matrixToText(invertedMatr.matrix);
+    bot.sendMessage(chatId, text);
+  }
+};
+
+const rangeOfMatr = (msg, match) => {
+  const chatId = msg.chat.id;
+  const name = match[1];
+  if (!IsThereMatr(chatId, name)) {
+    bot.sendMessage(chatId, answ.matrNotExist);
+  } else {
+    const matr = matrices[chatId][name];
+    const range = matr.getRange();
+    bot.sendMessage(chatId, `Range is ${range}`);
+  }
+};
+
+const matrToFract = matr => {
+  for (let i = 0; i < matr.length; i++) {
+    for (let j = 0; j < matr[0].length; j++) {
+      const el = Number(matr[i][j]);
+      if (!isNaN(el)) {
+        matr[i][j] = el;
+      }
+    }
+  }
+};
+
+const deleteEmptyStr = arr => {
+  for (let i = arr.length - 1; i >= 0; i--) {
+    if (arr[i] === '') {
+      arr.splice(i, 1);
+    }
+  }
+};
+
+const parseMatrix = text => {
+  const rows = text.split('\n');
+  const matr = rows.map(row => {
+    const splited = row.split(' ');
+    deleteEmptyStr(splited);
+    return splited.map(el => el.trim());
+  });
+  return matr;
+};
+
+const getKeyboardArr = name => [[`/print ${name}`],
+  [`/determinant ${name}`],
+  [`/range ${name}`],
+  [`/transpose ${name}`],
+  [`/invert ${name}`],
+  ['/help'],
+];
+
+const inputMatrix = (msg, name) => {
+  const chatId = msg.chat.id;
+  if (isComand(msg)) return;
+  const matr = parseMatrix(msg.text);
+  bot.sendMessage(chatId, `Is your matrix:\n${matrixToText(matr)}`);
+  if (Matrix.isValid(matr)) {
+    matrToFract(matr);
+    if (!matrices[chatId]) {
+      matrices[chatId] = {};
+    }
+    matrices[chatId][name] = new Matrix(matr);
+    bot.sendMessage(msg.chat.id, 'What do you want to do ?', {
+      'reply_markup': {
+        'keyboard': getKeyboardArr(name),
+      }
+    });
+  } else {
+    bot.sendMessage(chatId, answ.invalidMatr);
+    bot.once('message', msg => inputMatrix(msg, name));
+  }
+};
+
+
+const solveSoLE = msg => {
+  const chatId = msg.chat.id;
+  const rows = msg.text.split('\n');
+  const matr = rows.map(row => row.split(' '));
+  if (Matrix.isValid(matr)) {
+    matrToFract(matr);
+    bot.sendMessage(chatId, systemToText(matr));
+    const extendMatr = new Matrix(matr);
+    const solution = Matrix.findSolOfSoLE(extendMatr);
+    bot.sendMessage(chatId, solution);
+  } else {
+    bot.sendMessage(chatId, answ.invalidSystem);
+    bot.once('message', solveSoLE);
+  }
+};
+
+const inputSoLe = msg => {
+  const chatId = msg.chat.id;
+  bot.sendMessage(chatId, answ.enterSystem);
+  bot.once('message', solveSoLE);
+};
+
+const addMatrix = (msg, match) => {
+  const chatId = msg.chat.id;
+  const name = match[1];
+  bot.sendMessage(chatId, `${answ.sendName}${name}`);
+  bot.sendMessage(chatId, answ.enterMatr);
+  bot.once('message', msg => inputMatrix(msg, name));
+};
+
+bot.onText(/\/start/, msg => {
```

Comment:
This could use some better code separation, i.e. have commands folder where each file would describe 1 command and export appropriate function(s)/name and then gather and register them here.